resolver/retry_resolver.go

```diff
@@ -25,19 +25,29 @@ func NewRetryResolver(resolver Resolver, backoffs []time.Duration) Resolver {
 	}
 }
 
-func (r *retryResolver) retry(ctx context.Context, fn func() error) (err error) {
+func shouldRetry(err error) bool {
+	if err == context.DeadlineExceeded {
```

Comment:
What about `errors.Is(err, context.DeadlineExceeded)`?

Since (something I just learned)...

```go
e1 := context.DeadlineExceeded
e2 := fmt.Errorf("wrap context error: %w", e1)

fmt.Println(e2 == e1) // false
fmt.Println(errors.Is(e2, e1)) // true
```

Reply:
Ah, yes, good catch 👍